substrate/client/network/src/request_responses.rs

```diff
@@ -62,14 +62,84 @@ use std::{
 	sync::Arc,
 	task::{Context, Poll},
 	time::{Duration, Instant},
+	fmt,
 };
 
 pub use libp2p::request_response::{Config, InboundFailure, OutboundFailure, RequestId};
 
+/// Adding a custom OutboundFailure, not depending on libp2p
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+#[error("dial-failure")]
```

Comment:
```suggestion
```

Reply:
This was recommended by you to avoid implementing the `Display` trait. I removed it will give an error .

Reply:
I was wrong, `thiserror` does not support converting docstrings into error `Display` impl. At least it is not mentioned in the [docs](https://docs.rs/thiserror/latest/thiserror/).

`#[error("Error description ...")]` should be defined for every enum variant for them to provide different messages.